src/Features/Core/Portable/SyncNamespaces/AbstractSyncNamespacesSevice.cs

```diff
@@ -0,0 +1,171 @@
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CodeFixes.MatchFolderAndNamespace;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.SyncNamespaces
+{
+    internal abstract class AbstractSyncNamespacesSevice<TSyntaxKind, TNamespaceSyntax>
+        : ISyncNamespacesService
+        where TSyntaxKind : struct
+        where TNamespaceSyntax : SyntaxNode
+    {
+        public abstract AbstractMatchFolderAndNamespaceDiagnosticAnalyzer<TSyntaxKind, TNamespaceSyntax> DiagnosticAnalyzer { get; }
+        public abstract AbstractChangeNamespaceToMatchFolderCodeFixProvider CodeFixProvider { get; }
+
+        public async Task<Solution> SyncNamespacesAsync(
+            ImmutableArray<Project> projects,
+            CancellationToken cancellationToken)
+        {
+            var solution = projects[0].Solution;
+            var diagnosticAnalyzers = ImmutableArray.Create<DiagnosticAnalyzer>(DiagnosticAnalyzer);
+            var diagnosticsByProject = await GetDiagnosticsByProjectAsync(projects, diagnosticAnalyzers, cancellationToken).ConfigureAwait(false);
+
+            // If no diagnostics are reported, then there is nothing to fix.
+            if (diagnosticsByProject.Values.All(diagnostics => diagnostics.IsEmpty))
+            {
+                return solution;
+            }
+
+            var fixAllContext = await GetFixAllContextAsync(solution, CodeFixProvider, diagnosticsByProject, cancellationToken).ConfigureAwait(false);
+            var fixAllProvider = CodeFixProvider.GetFixAllProvider();
+            RoslynDebug.AssertNotNull(fixAllProvider);
+
+            return await ApplyCodeFixAsync(fixAllProvider, fixAllContext, cancellationToken).ConfigureAwait(false);
+        }
+
+        private static async Task<ImmutableDictionary<Project, ImmutableArray<Diagnostic>>> GetDiagnosticsByProjectAsync(
+            ImmutableArray<Project> projects,
+            ImmutableArray<DiagnosticAnalyzer> diagnosticAnalyzers,
+            CancellationToken cancellationToken)
+        {
+            var builder = ImmutableDictionary.CreateBuilder<Project, ImmutableArray<Diagnostic>>();
+
+            foreach (var project in projects)
+            {
+                var diagnostics = await GetDiagnosticsAsync(project, diagnosticAnalyzers, cancellationToken).ConfigureAwait(false);
+                builder.Add(project, diagnostics);
+            }
+
+            return builder.ToImmutable();
+        }
+
+        private static async Task<ImmutableArray<Diagnostic>> GetDiagnosticsAsync(
+            Project project,
+            ImmutableArray<DiagnosticAnalyzer> diagnosticAnalyzers,
+            CancellationToken cancellationToken)
+        {
+            var compilation = await project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+            RoslynDebug.AssertNotNull(compilation);
+
+            var analyzerOptions = new CompilationWithAnalyzersOptions(
```

Comment:
Will this work if this is disabled in the editorconfig?

Reply:
Yes because I am specifically running the analyzer and ignoring the returned severity.